core/src/databases/database.rs

```diff
@@ -58,16 +58,23 @@ impl Database {
         match self.db_type {
             DatabaseType::REMOTE => Err(anyhow!("Remote DB not implemented.")),
             DatabaseType::LOCAL => {
-                let rows = self.get_rows(store).await?;
-
-                rows.par_iter()
-                    .map(|(table, r)| {
-                        Ok(DatabaseSchemaTable::new(
-                            table.clone(),
-                            TableSchema::from_rows(&r)?,
-                        ))
+                let (tables, _) = store
+                    .list_databases_tables(
+                        &self.project,
+                        &self.data_source_id,
+                        &self.database_id,
+                        None,
+                    )
+                    .await?;
+
+                Ok(tables
+                    .iter()
+                    // Ignore empty tables.
+                    .filter_map(|t| match t.schema() {
+                        None => None,
+                        Some(s) => Some(DatabaseSchemaTable::new(t.clone(), s.clone())),
```

Comment:
You probably don't need to clone `t` if you use `into_iter()` ?

Reply:
Thanks I was able to remove the `clone` here. However it means that `t.schema()` has to clone.
But it still saves one clone 👍

Reply:
(outdated anyway since we no longer need this code thanks to removing DatabaseSchemaTable)